airflow/providers/asana/operators/asana_tasks.py

```diff
@@ -0,0 +1,175 @@
+        You must specify at least one of 'workspace', 'parent', or 'projects'
+        either here or in the connection.
+    :type task_parameters: dict
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        conn_id: str,
+        name: str,
+        task_parameters: Optional[dict] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.conn_id = conn_id
+        self.name = name
+        self.task_parameters = task_parameters
+
+    def execute(self, context: Dict) -> str:
+        hook = AsanaHook(conn_id=self.conn_id)
+        response = hook.create_task(self.name, self.task_parameters)
+        self.log.info(response)
+        return response["gid"]
+
+
+class AsanaUpdateTaskOperator(BaseOperator):
+    """
+    This operator can be used to update Asana tasks.
+    For more information on Asana optional task parameters, see
+    https://developers.asana.com/docs/update-a-task
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AsanaUpdateTaskOperator`
+
+    :param conn_id: The Asana connection to use.
+    :type conn_id: str
+    :param asana_task_gid: Asana task ID to update
+    :type asana_task_gid: str
+    :param task_parameters: Any task parameters that should be updated.
+        See https://developers.asana.com/docs/update-a-task for a complete list.
+    :type task_update_parameters: dict
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        conn_id: str,
+        asana_task_gid: str,
+        task_parameters: dict,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.conn_id = conn_id
+        self.asana_task_gid = asana_task_gid
+        self.task_parameters = task_parameters
+
+    def execute(self, context: Dict) -> None:
+        hook = AsanaHook(conn_id=self.conn_id)
+        response = hook.update_task(self.asana_task_gid, self.task_parameters)
+        self.log.info(response)
+
+
+class AsanaDeleteTaskOperator(BaseOperator):
+    """
+    This operator can be used to delete Asana tasks.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AsanaDeleteTaskOperator`
+
+    :param conn_id: The Asana connection to use.
+    :type conn_id: str
+    :param asana_task_gid: Asana Task ID to delete.
+    :type asana_task_gid: str
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        conn_id: str,
+        asana_task_gid: str,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.conn_id = conn_id
+        self.asana_task_gid = asana_task_gid
+
+    def execute(self, context: Dict) -> None:
+        hook = AsanaHook(conn_id=self.conn_id)
+        response = hook.delete_task(self.asana_task_gid)
```

Comment:
What will happen if the task does not exist?

Reply:
I now catch the exception in this case and log that the task id wasn't found. Thanks again for all the feedback/help!